src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -898,6 +898,12 @@ - (void)_triggerResubscribeWithReason:(NSString *)reason nodeLikelyReachable:(BO
             subscriptionCallback->ResetResubscriptionBackoff();
         }
         readClientToResubscribe->TriggerResubscribeIfScheduled(reason.UTF8String);
+    } else if (_internalDeviceState == MTRInternalDeviceStateSubscribing && nodeLikelyReachable) {
+        // If we have reason to suspect that the node is now reachable and we haven’t established a
+        // CASE session yet, let’s consider it to be stalled and invalidate the pairing session.
+        dispatch_async(self.queue, ^{
+            [[self _concreteController] invalidateCASESessionEstablishmentForNode:self->_nodeID];
```

Comment:
We shouldn't add new controller API here, I would think.  Just `asyncGetCommissionerOnMatterQueue` and go from there.

That said, do we have any indication that we are not just going to get a slew of _triggerResubscribeWithReason from DNS-SD stuff while nothing has really changed?  I'm really worried about us getting into a state where we just keep canceling things and never manage to subscribe.  It does help that we only do this when we have no CASE session yet, at least....

Reply:
https://github.com/project-chip/connectedhomeip/pull/36315